core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestSubscriber.java

```diff
@@ -111,6 +113,13 @@ public void onComplete() {
 
     @Override
     void onWriteSuccess() {
+        if (needs100Continue) {
```

Comment:
optional; I was a little surprised about this until I realized `needs100Continue` is mutable. Is it possible to remove `needs100Continue` and rely on checking `NEEDS_100_CONTINUE` instead?

In general, I prefer less states are maintained if possible unless there is a good reason to do so.

```suggestion
        if (state() == State.NEEDS_100_CONTINUE) {
```

Reply:
That's a good suggestion. Addressed. 👍